[Docs] Rewrite explore section of docs using yugabyted (#8678)
yugabyte/yugabyte-db: docs/content/latest/explore/linear-scalability/sharding-data.md

## Try it out

In this tutorial, you'll explore automatic sharding inside YugabyteDB. First, you'll create some tables to understand how automatic sharding works. Then, you'll insert entries one by one, and examine how the data gets distributed across the various nodes.

This tutorial uses the [yugabyted](../../../reference/configuration/yugabyted) cluster management utility.

### Create a universe

To create a universe, do the following:

1. Let’s begin by creating a single node cluster. Run the following command.
   ```sh
   $ ./bin/yugabyted start \
                     --base_dir=/tmp/ybd1 \
                     --listen=127.0.0.1 \
                     --tserver_flags "memstore_size_mb=1"

>    ```

Reviewer: Space after fenced code block
````suggestion
   ```

````

   * `memstore_size_mb=1` sets the total size of memstores on the tablet-servers to `1MB`. This will force a flush of the data to disk when a value greater than 1MB is added, so that you can observe which tablets the data is written to.

1. Add two more nodes to make this a 3-node by joining them with the previous node. We need to pass the memstore_size flag to each of the added YB-TServer servers.

    ```sh
    $ ./bin/yugabyted start \
                      --base_dir=/tmp/ybd2 \
                      --listen=127.0.0.2 \
                      --join=127.0.0.1 \
                      --tserver_flags "memstore_size_mb=1"
    ```

    ```sh
    $ ./bin/yugabyted start \
                      --base_dir=/tmp/ybd3 \
                      --listen=127.0.0.3 \
                      --join=127.0.0.1 \
                      --tserver_flags "memstore_size_mb=1"
    ```

> {{< note title="Warning" >}}

Reviewer: Space after fenced code block, and admonition should be a warning

```suggestion

{{< warning title="Warning" >}}
```

Setting `memstore_size` to such a low value is not recommended in production, and is only being used here to illustrate the point by forcing flushes to happen more quickly.

> {{< /note >}}

Reviewer: ```suggestion
{{< /warning >}}
```

ycqlsh> CREATE TABLE ybdemo_keyspace.cassandrakeyvalue (k text PRIMARY KEY, v blob);
```

By default, [yugabyted](../../../reference/configuration/yugabyted) creates one tablet per node per table. So for a 3 node cluster, 3 tablets are created for the above table; one on every node. Every such tablet is replicated 3 times for fault tolerance, so that makes the total number of nodes to be 3*3=9. Every node thus contains 3 tablets, one of which it is the leader and the remaining 2 of which it is the follower. 

### Explore tablets

* **The tablets are evenly balanced across the various nodes**. You can see the number of tablets per node in the Tablet Servers page of the master Admin UI, by going to the [table details page](http://127.0.0.1:7000/tablet-servers). The page should look something like the image below.

> ![Number of tablets in the table](/images/ce/sharding_evenload.png)

Reviewer: Let's indent these things
```suggestion
    ![Number of tablets in the table](/images/ce/sharding_evenload.png)
```

> Notice that each node has 3 tablets, and the total number of tablets is 9 as expected. Out of these 3, it is the leader of 1 and follower of other 2.

Reviewer: ```suggestion
    Notice that each node has 3 tablets, and the total number of tablets is 9 as expected. Out of these 3, it is the leader of 1 and follower of other 2.
```

* **The table has 3 shards, each owning a range of the keyspace**. Navigate to the [table details page](http://127.0.0.1:7000/table?keyspace_name=ybdemo_keyspace&table_name=cassandrakeyvalue) to examine the various tablets. This page should look as follows.

> ![Tablet details of the table](/images/ce/sharding_keyranges.png)

Reviewer: ```suggestion
    ![Tablet details of the table](/images/ce/sharding_keyranges.png)
```

> There are 3 shards as expected, and the key ranges owned by each tablet are shown. This page also shows which node is currently hosting (and is the leader for) each tablet. Note here that tablet balancing across nodes happens on a per-table basis, with each table scaled out to an appropriate number of nodes.

Reviewer: ```suggestion
    There are 3 shards as expected, and the key ranges owned by each tablet are shown. This page also shows which node is currently hosting (and is the leader for) each tablet. Note here that tablet balancing across nodes happens on a per-table basis, with each table scaled out to an appropriate number of nodes.
```

* **Each tablet has a separate directory dedicated to it for data**. List out all the tablet directories and check their sizes, as follows:

1. First get the table-id of the table we created by going to the [table listing page](http://127.0.0.1:7000/tables) and accessing the row corresponding to `ybdemo_keyspace.cassandrakeyvalue`. In this illustration, the table-id is `769f533fbde9425a8520b9cd59efc8b8`.

> ![Id of the created table](/images/ce/sharding_tableid.png)

Reviewer: indent the image, and the auto-numbering will work right

```suggestion
    ![Id of the created table](/images/ce/sharding_tableid.png)
```

> 2. Next, we can view all the tablet directories and their sizes for this table by running the following command. Remember to replace the id with your corresponding id.

Reviewer: ```suggestion
1. Next, you can view all the tablet directories and their sizes for this table by running the following command. Remember to replace the id with your corresponding id.
```

```sh
$ du -hs /tmp/ybd*/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/* | grep -v '0B'

> ```

Reviewer: Indent this too

> ```output

Reviewer: also indent this block

 28K	/tmp/ybd1/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-0149071638294c5d9328c4121ad33d23
 28K	/tmp/ybd1/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-0df2c7cd87a844c99172ea1ebcd0a3ee
 28K	/tmp/ybd1/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-59b7d53724944725a1e3edfe9c5a1440
 28K	/tmp/ybd2/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-0149071638294c5d9328c4121ad33d23
 28K	/tmp/ybd2/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-0df2c7cd87a844c99172ea1ebcd0a3ee
 28K	/tmp/ybd2/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-59b7d53724944725a1e3edfe9c5a1440
 28K	/tmp/ybd3/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-0149071638294c5d9328c4121ad33d23
 28K	/tmp/ybd3/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-0df2c7cd87a844c99172ea1ebcd0a3ee
 28K	/tmp/ybd3/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-59b7d53724944725a1e3edfe9c5a1440
 ```

> We can see that there are 9 nine entries - one corresponding to each tablet with 28K bytes as the size.

Reviewer: Indent, and add space following

```suggestion
    There are nine entries, one corresponding to each tablet with 28K bytes as the size.

```

1. Download the YugabyteDB workload generator JAR file (`yb-sample-apps.jar`):

    ```sh

>     $ wget https://github.com/yugabyte/yb-sample-apps/releases/download/1.3.3/yb-sample-apps.jar

Reviewer: Did you lower the version number on purpose here?

Author: It wasn't working with 1.3.6 for some reason.

    28K	/tmp/ybd2/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-0149071638294c5d9328c4121ad33d23
    9.6M    /tmp/ybd1/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-0df2c7cd87a844c99172ea1ebcd0a3ee
    28K	/tmp/ybd1/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-59b7d53724944725a1e3edfe9c5a1440
    28K	/tmp/ybd2/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-0149071638294c5d9328c4121ad33d23
    9.6M    /tmp/ybd2/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-0df2c7cd87a844c99172ea1ebcd0a3ee
    28K	/tmp/ybd2/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-59b7d53724944725a1e3edfe9c5a1440
    28K	/tmp/ybd3/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-0149071638294c5d9328c4121ad33d23
    9.6M    /tmp/ybd3/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-0df2c7cd87a844c99172ea1ebcd0a3ee
    28K	/tmp/ybd3/data/yb-data/tserver/data/rocksdb/table-769f533fbde9425a8520b9cd59efc8b8/tablet-59b7d53724944725a1e3edfe9c5a1440
    ```

> We can see that the key was successfully inserted in one of the tablets leading to a proliferation of size to 9.6 MB. Because this tablet has 3 copies distributed across the 3 nodes, we can see 3 entries of this size.

Reviewer: Space after code block
```suggestion

You can see that the key was successfully inserted in one of the tablets leading to a proliferation of size to 9.6 MB. Because this tablet has 3 copies distributed across the 3 nodes, there are 3 entries of this size.
```

    ```sh
    $ ./bin/yugabyted start \
                      --base_dir=/tmp/ybd4 \
                      --listen=127.0.0.4 \
                      --join=127.0.0.1 \
                      --tserver_flags "memstore_size_mb=1"
    ```

1. Check the tablet servers page, to see that the tablets are re-distributed evenly among the 4 nodes:

    ![Auto-sharding when adding one node](/images/ce/sharding_4nodes.png)

1. Add 2 more nodes to the universe, making it a total of 6 nodes:

    ```sh
    $ ./bin/yugabyted start \
                      --base_dir=/tmp/ybd5 \
                      --listen=127.0.0.5 \
                      --join=127.0.0.1 \
                      --tserver_flags "memstore_size_mb=1"
    ```

    ```sh
    $ ./bin/yugabyted start \
                      --base_dir=/tmp/ybd5 \
                      --listen=127.0.0.5 \
                      --join=127.0.0.1 \
                      --tserver_flags "memstore_size_mb=1"
    ```

1. Verify that the tablets are evenly distributed across the 6 nodes. Each node now has 2 tablets.

    ![Auto-sharding when adding three nodes](/images/ce/sharding_6nodes.png)

### [Optional] Clean up

> If you're done experimenting, run the following command to shut down the local cluster:

Reviewer: ```suggestion
If you're done experimenting, run the following commands to shut down the local cluster:
```

> ```sh

Reviewer: You could combine all the shell commands into one block...

Author: It brings more clarity I believe by saying that we are shutting down nodes individually.
